src/ai/assistant.ts

```diff
@@ -0,0 +1,440 @@
+import _ from 'lodash';
+
+import FactoryBase from '../factoryBase';
+import AinftObject from '../ainft721Object';
+import { AinizeService, OperationType } from '../ainize';
+import {
+  Assistant,
+  AssistantCreateParams,
+  AssistantDeleteTransactionResult,
+  AssistantDeleted,
+  AssistantMinted,
+  AssistantTransactionResult,
+  AssistantUpdateParams,
+  NftToken,
+  NftTokens,
+  QueryParams,
+} from '../types';
+import {
+  MESSAGE_GC_MAX_SIBLINGS,
+  MESSAGE_GC_NUM_SIBLINGS_DELETED,
+  THREAD_GC_MAX_SIBLINGS,
+  THREAD_GC_NUM_SIBLINGS_DELETED,
+} from '../constants';
+import {
+  buildSetTxBody,
+  buildSetValueOp,
+  sendTx,
+  buildSetWriteRuleOp,
+  buildSetOp,
+  buildSetStateRuleOp,
+  getChecksumAddress,
+} from '../utils/util';
+import {
+  isObjectOwner,
+  validateAssistant,
+  validateDuplicateAssistant,
+  validateObject,
+  validateServerConfigurationForObject,
+  validateToken,
+} from '../utils/validator';
+import { Path } from '../utils/path';
+
+/**
+ * This class supports building assistants that enables conversation with LLM models.\
+ * Do not create it directly; Get it from AinftJs instance.
+ */
+export class Assistants extends FactoryBase {
+  private ainize: AinizeService = AinizeService.getInstance();
+
+  /**
+   * Create an assistant with a model and instructions.
+   * @param {string} objectId - The ID of AINFT object.
+   * @param {string} tokenId - The ID of AINFT token.
+   * @param {AssistantCreateParams} AssistantCreateParams - The parameters to create assistant.
+   * @returns Returns a promise that resolves with both the transaction result and the created assistant.
+   */
+  async create(
+    objectId: string,
+    tokenId: string,
+    { model, name, instructions, description, metadata }: AssistantCreateParams
+  ): Promise<AssistantTransactionResult> {
+    const address = this.ain.signer.getAddress();
+
+    await validateObject(this.ain, objectId);
+    const isAdmin = await isObjectOwner(this.ain, objectId, address);
+    await validateToken(this.ain, objectId, tokenId);
+    await validateDuplicateAssistant(this.ain, objectId, tokenId);
+
+    const serverName = this.ainize.getServerName();
+    await validateServerConfigurationForObject(this.ain, objectId, serverName);
+
+    const opType = OperationType.CREATE_ASSISTANT;
+    const body = {
+      model,
+      name,
+      instructions,
+      ...(description && { description }),
+      ...(metadata && !_.isEmpty(metadata) && { metadata }),
+    };
+
+    const { data } = await this.ainize.requestWithAuth<Assistant>(this.ain, {
+      serverName,
+      opType,
+      data: body,
+    });
+
+    if (!isAdmin) {
```

Comment:
@platfowner if user is not admin, only response data is returned. (in this case, backend server do admin operation)
but if user is admin, they send transaction with response data.